File: A4.Empower-Git/A4.Empower/A4.BAL/Performance/RatingModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace A4.BAL
{
    public class RatingModel
    {
        public RatingModel()
        {
            RatingList = new List<Ratings>();
            MidYearRating = new Rating();
            EndYearRating = new Rating();
            CheckSaveSubmit = new CheckSaveSubmit();
        }
        public string InstructionText { get; set; }
        public List<Ratings> RatingList { get; set; }
        public Rating MidYearRating;
        public Rating EndYearRating;
        public CheckSaveSubmit CheckSaveSubmit;
    }
    
    public class Rating
    {
        public string RatingId;
        public string ManagerComment;
        public string ManagerSignature;
        public string EmployeeComment;
    }

    public class Ratings
    {
        public string RatingId { get; set; }
        public string RatingName { get; set; }
        public string RatingDescription { get; set; }


    }
}

File: ClientApp/src/app/models/performance/common/rating.model.ts
import { CheckSaveSubmit } from "./check-save-submit.model";

export class RatingModel {
    constructor() {
        this.checkSaveSubmit = new CheckSaveSubmit();
        this.ratingList = new Array<Ratings>();
    }
    public instructionText: string;
    public ratingList: Ratings[];
    public midYearRating: Rating;
    public endYearRating: Rating;
    public checkSaveSubmit: CheckSaveSubmit;
}
export class Rating {
    public ratingId: string;
    public managerComment: string;
    public managerSignature: string;
    public employeeComment: string;

}

export class Ratings {
    public ratingId: string;
    public ratingName: string;
    public ratingDescription: string;
}